src/Tmds.Ssh/SshConnection.cs

```diff
@@ -22,6 +22,8 @@ protected SshConnection(SequencePool sequencePool)
 
     public abstract ValueTask<Packet> ReceivePacketAsync(CancellationToken ct, int maxLength = Constants.PreAuthMaxPacketLength);
     public abstract ValueTask SendPacketAsync(Packet packet, CancellationToken ct);
-    public abstract void Dispose();
     public abstract void SetEncryptorDecryptor(IPacketEncryptor packetEncoder, IPacketDecryptor packetDecoder, bool resetSequenceNumbers, bool throwIfReceiveSNZero);
+
+    public void Dispose() => Dispose(true);
+    protected virtual void Dispose(bool isDisposing) {}
```

Comment:
I don't use this pattern unless there are native resources involved (in this type or derived ones).

Reply:
After the removal of the specialised Socket class this is no longer needed anyway but will keep it in mind for the future.